### ISSUE TITLE
libluv fails to build on macOS

### ISSUE THREAD

**Issue by certik** [LLNL/spack#22594]
Same setup as in #22593.

```
==> Installing libluv-1.36.0-0-haxhwex2jnr4uukuhyd2ud6ju676vmbd
==> No binary for libluv-1.36.0-0-haxhwex2jnr4uukuhyd2ud6ju676vmbd found: installing from source
==> Fetching https://spack-llnl-mirror.s3-us-west-2.amazonaws.com/_source-cache/archive/f2/f2e7eb372574f25c6978c1dc74280d22efdcd7df2dda4a286c7fe7dceda26445.tar.gz
######################################################################### 100.0%
==> No patches needed for libluv
==> libluv: Executing phase: 'cmake'
==> libluv: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j8'

5 errors found in build log:
     71     /Users/certik/repos/spack/opt/spack/darwin-catalina-skylake/apple-c
            lang-12.0.0/cmake-3.20.0-awb3d4di2lo2h4axkavqfbd6gnx355v5/bin/cmake
             -P CMakeFiles/libluv_a.dir/cmake_clean_target.cmake
     72     /Users/certik/repos/spack/lib/spack/env/clang/clang -O2 -g -DNDEBUG
             -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk
             -bundle -Wl,-headerpad_max_install_names -flat_namespace -undefine
            d suppress -o luv.so CMakeFiles/luv.dir/src/luv.c.o  /Users/certik/
            repos/spack/opt/spack/darwin-catalina-skylake/apple-clang-12.0.0/li
            buv-1.40.0-hedlhczvlecsskuiucquxkpb6hnodxjf/lib/libuv.dylib
     73     /Users/certik/repos/spack/opt/spack/darwin-catalina-skylake/apple-c
            lang-12.0.0/cmake-3.20.0-awb3d4di2lo2h4axkavqfbd6gnx355v5/bin/cmake
             -E cmake_link_script CMakeFiles/libluv_a.dir/link.txt --verbose=1
     74     /usr/bin/ar qc libluv_a.a CMakeFiles/libluv_a.dir/src/luv.c.o
     75     /usr/bin/ranlib libluv_a.a
     76     [100%] Built target libluv_a
  >> 77     Undefined symbols for architecture x86_64:
     78       "_luaL_argerror", referenced from:
     79           _luv_cancel in luv.c.o
     80           _luv_timer_start in luv.c.o
     81           _luv_timer_stop in luv.c.o
     82           _luv_timer_again in luv.c.o
     83           _luv_timer_set_repeat in luv.c.o

     ...

     469          _luv_thread_arg_set in luv.c.o
     470          _luv_check_bufs in luv.c.o
     471          _luv_prep_bufs in luv.c.o
     472          _luv_check_bufs_noref in luv.c.o
     473          ...
     474    ld: symbol(s) not found for architecture x86_64
  >> 475    clang: error: linker command failed with exit code 1 (use -v to see
             invocation)
  >> 476    make[2]: *** [libluv.1.36.0.dylib] Error 1
  >> 477    make[1]: *** [CMakeFiles/libluv.dir/all] Error 2
     478    make[1]: *** Waiting for unfinished jobs....
     479    [100%] Built target luv
  >> 480    make: *** [all] Error 2

See build log for details:
  /var/folders/_w/414v9g5d59gfc7f8qmgwl904000fsc/T/certik/spack-stage/spack-stage-libluv-1.36.0-0-haxhwex2jnr4uukuhyd2ud6ju676vmbd/spack-build-out.txt
```

**Comment by certik** [comment]
Duplicate of #26673.

**Comment by certik** [comment]
This was fixed by #29228.
